src/CoreWCF.Primitives/src/CoreWCF/IdentityModel/Tokens/TokenReplayCache.cs

```diff
@@ -0,0 +1,48 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using CoreWCF.IdentityModel.Configuration;
+using Microsoft.IdentityModel.Tokens;
+using System;
+using System.Xml;
+
+namespace CoreWCF.IdentityModel.Tokens
+{
+    /// <summary>
+    /// This class defines the API for a cache that stores tokens for and purges them 
+    /// on a schedule time interval.
+    /// </summary>
+    public abstract class TokenReplayCache : ITokenReplayCache
+    {
+        /// <summary>
+        /// Attempt to add a new entry or update an existing entry.
+        /// </summary>
+        /// <param name="key">Key to use when adding item</param>
+        /// <param name="securityToken">SecurityToken to add to cache, can be null</param>
+        /// <param name="expirationTime">The expiration time of the entry.</param>
+       // public abstract void AddOrUpdate(string key, SecurityToken securityToken, DateTime expirationTime);
+
```

Comment:
Why did you comment out AddOrUpdate?

Reply:
we were not using anywhere... TryAdd is replacement and supported inside IM(MS)